guac_core/src/counterparty.rs

```diff
@@ -3,6 +3,7 @@ use althea_types::EthAddress;
 #[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, Hash)]
 pub struct Counterparty {
     pub address: EthAddress,
+    // assuming ipv6 socketaddr
```

Comment:
Why do we have to make this assumption? Isn't `guac` meant to be universal and usable outside of Althea?

Reply:
the problem is that the actix http client acts weird when given hostnames which are ip addressed and there is no internet. Going to dig into that at some point but for now, I just manually create a TCP socket for the http requests to go over, which means we don't do any hostname resolution and only work with ipv6 sockeraddrs.